soccer/src/soccer/strategy/agent/position/offense.cpp

```diff
@@ -37,7 +36,7 @@ Offense::State Offense::next_state() {
 
         case SEEKING: {
             // If the ball seems "stealable", we should switch to STEALING
-            if (can_steal_ball()) {
+            if (can_steal_ball() && !ball_in_red(last_world_state_)) {
```

Comment:
incorporate the `ball_in_red` into `can_steal_ball()` they can be separate functions but have `can_steal_ball` check `ball_in_red` directly